src/foundry/client/documents/abstract/world-collection.d.mts

```diff
@@ -74,122 +71,205 @@ declare abstract class WorldCollection<
+    | (DocumentName extends "Scene" ? collections.Scenes.Implementation : never)
+    | (DocumentName extends "Setting" ? collections.WorldSettings.Implementation : never)
+    | (DocumentName extends "User" ? collections.Users.Implementation : never);
+
+  /**
+   * This type exists because there are {@link Document.WorldType | world documents}
+   * that are not valid {@link CONST.FOLDER_DOCUMENT_TYPES | Folder types}.
+   */
   type Folders<DocumentName extends Document.WorldType> = Collection<
     DocumentName extends Folder.DocumentType ? Folder.Stored<DocumentName> : never
   >;
 
-  type Directory<DocumentName extends Document.WorldType, Name extends string> =
-    Lowercase<Name> extends keyof typeof ui
-      ? (typeof ui)[Lowercase<Name>]
-      :
-          | (DocumentName extends foundry.CONST.FOLDER_DOCUMENT_TYPES
-              ? DocumentDirectory<Document.ImplementationClassFor<DocumentName>>
-              : never)
-          | AbstractSidebarTab.Any
-          | undefined
-          | null;
-
-  interface FromCompendiumOptions {
+  /**
+   * {@linkcode WorldCollection.directory | WorldCollection#directory} returns `ui[documentClass.metadata.collection]`, but several
+   * subclasses have overrides:
+   * - {@linkcode collections.ChatMessages.directory | ChatMessages#directory} returns {@linkcode ChatLog | ui.chat}, notably *not*
+   * extending {@linkcode DocumentDirectory}, which is what `WorldCollection#directory` is typed as by Foundry.
+   * - {@linkcode collections.CombatEncounters.directory | CombatEncounters#directory} returns {@linkcode CombatTracker | ui.combat}, also
+   * not extending `DocumentDirectory`.
+   * - {@linkcode collections.RollTables.directory | RollTables#directory} just uses a different key than its `metadata.collection`, and
+   * {@linkcode collections.Macros.directory | Macros#directory} unnecessarily returns `ui.macros`, which is what it would be anyway, but
+   * these are both at least `DocumentDirectory`s.
+   * - {@linkcode collections.WorldSettings.directory | WorldSettings#directory} returns `null`.
+   *
+   * All but the last are only guaranteed by the `ready` hook, which is baked into `typeof ui`.
+   */
+  type Directory<DocumentName extends Document.WorldType> = DocumentName extends "Setting"
+    ? null
+    : DocumentName extends "ChatMessage"
+      ? typeof ui.chat // instead of `messages`
+      : DocumentName extends "Combat"
+        ? typeof ui.combat // instead of `combats`
+        : _Directory<DocumentName, Document.MetadataFor<DocumentName>["collection"]>;
+
+  /**
+   * The fallback case is:
+   * - `DocumentDirectory` for the doc name, both because if `ui` gets borked its accurate-ish for existing directories, and because that's
+   * what Foundry types {@linkcode WorldCollection.directory | WorldCollection#directory} as, exceptions notwithstanding.
+   * - `| undefined` instead of `InitializedOn<..., "ready">` because {@linkcode collections.Users | Users} and
```

Comment:
Are there cases where it'd be useful to type with `InitializedOn`? Is it worth bothering?